instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/OtelResourceAutoConfiguration.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.spring.autoconfigure;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.sdk.extension.resources.ContainerResource;
+import io.opentelemetry.sdk.extension.resources.HostResource;
+import io.opentelemetry.sdk.extension.resources.OsResource;
+import io.opentelemetry.sdk.extension.resources.ProcessResource;
+import io.opentelemetry.sdk.extension.resources.ProcessRuntimeResource;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
+import java.util.Map;
+import java.util.function.Supplier;
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.Environment;
+
+@Configuration
+@EnableConfigurationProperties(OtelResourceProperties.class)
+@AutoConfigureBefore(OpenTelemetryAutoConfiguration.class)
+@ConditionalOnProperty(prefix = "otel.springboot.resource", name = "enabled", matchIfMissing = true)
+public class OtelResourceAutoConfiguration {
+
+  @Bean
+  public Supplier<Resource> otelResourceProvider(
+      Environment env, OtelResourceProperties otelResourceProperties) {
+    return () -> {
+      AttributesBuilder attributesBuilder = Attributes.builder();
+      for (Map.Entry<String, String> entry : otelResourceProperties.getAttributes().entrySet()) {
+        attributesBuilder.put(entry.getKey(), entry.getValue());
+      }
+      String otelServiceName = env.getProperty("otel.springboot.resource.attributes.service.name");
+      if (otelServiceName != null) {
+        attributesBuilder.put(ResourceAttributes.SERVICE_NAME, otelServiceName);
+      }
```

Comment:
Unfortunately, `otel.springboot.resource.attributes.service.name`  does not set name into attributes
`dot` between service.name does not allow parse name properly

Reply:
that's too bad, this will be a problem for lots of other resource attributes https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/resource/semantic_conventions#service

do you have a specific use-case or need for including generic resource attributes?

if not, maybe we should skip that until we have a solution that will generalize for all resource attributes

Reply:
I verified again and I was wrong
https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-Configuration-Binding#map-based-binding 
it is possible to define attributes with dotes, thank you @mateuszrzeszutek 
I added test for such cases (it seems I wrote wrong test before and it was a cause of wrong understanding)

Reply:
Awesome, thanks!